R/install.R

```diff
@@ -111,32 +133,25 @@ repository <-
         version = BiocManager::version(),
         binary_base_url = BINARY_BASE_URL)
 {
-    ## are we running on a docker container?
-    platform <- ""
-    bioconductor_docker_version <- Sys.getenv("BIOCONDUCTOR_DOCKER_VERSION")
-    if (!nzchar(bioconductor_docker_version)) {
-        platform <- Sys.getenv("TERRA_R_PLATFORM")
-        bioconductor_docker_version <-
-            Sys.getenv("TERRA_R_PLATFORM_BINARY_VERSION")
-    }
-    if (!nzchar(bioconductor_docker_version))
-        return(character())
+    platform_docker <- .platform_bioc_docker_version()
 
-    ## is the docker container configured correctly?
+    bioconductor_docker_version <-
+        platform_docker[["bioconductor_docker_version"]]
+    platform <- platform_docker[["platform"]]
     bioconductor_version <- package_version(version)
-    docker_version <- package_version(bioconductor_docker_version)
-    test <-
-        (bioconductor_version$major == docker_version$major) &
-        (bioconductor_version$minor == docker_version$minor)
-    if (!test) {
+
+    if (!nzchar(bioconductor_docker_version) ||
+        !.test_container_bioc_versions(
+            bioconductor_version, bioconductor_docker_version
+        )
+    )
```

Comment:
Please separate the `test` from the `if()` as in the original, so that (a) it is possible when debugging to know the `test` value before stepping into the conditional and (b) the code is easier to understand, because the test condition and conditional evaluation are separated, and the formatting is more straight-forward.